mirgecom/artificial_viscosity.py

```diff
@@ -152,6 +159,25 @@ def _facial_flux_r(discr, r_tpair):
     return discr.project(r_tpair.dd, "all_faces", flux_out)
 
 
+class AVBoundaryInterface(metaclass=ABCMeta):
+    r"""Abstract interface to boundary treatments used by artificial viscosity.
+
+    This is the interface that must be implemented by any boundary treatment
+    device used by the artificial viscosity operator (:func:`av_operator`).
+
+    .. automethod:: get_scalar_flux
+    .. automethod:: get_diffusion_flux
+    """
+
+    @abstractmethod
+    def get_scalar_flux(self, discr, btag, u, **kwargs):
+        """Get the scalar flux of *u* components through boundary faces in *btag*."""
+
+    @abstractmethod
+    def get_diffusion_flux(self, discr, btag, r, **kwargs):
+        """Get the flux of *r*=D(u)grad(u) through faces in *btag*."""
```

Comment:
I see I have messed up the interface to the diffusion operator.  I'm uncertain how to fix it just now since the native AV operator doesn't pass this stuff separate.  The docstring I'm less certain about.   The native operator seems to be calculating flux of *r*.

Reply:
> I see I have messed up the interface to the diffusion operator. I'm uncertain how to fix it just now since the native AV operator doesn't pass this stuff separate. The docstring I'm less certain about.

I think this is fine right now, it will just need to be changed when incorporating diffusion-based AV.

> The native operator seems to be calculating flux of _r_.

"flux r" or "diffusive flux of u" would make more sense I think.

Reply:
I definitely see your point, however in this particular case, the routine never actually deals with *u*, it deals with *r* and fluxes *r*.